ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -184,7 +184,7 @@ private void clean(CompactionInfo ci, long minOpenTxnGLB, boolean metricsEnabled
       if (metricsEnabled) {
         perfLogger.perfLogBegin(CLASS_NAME, cleanerMetric);
       }
-      Table t = resolveTable(ci);
+      Table t = cacheAndResolveTable(ci);
```

Comment:
what do you think if we put `cache` at the end and keep the primary goal at the beginning? 
````
resolveTableAndCache
````

Reply:
Sounds reasonable, fixed.